### ISSUE TITLE
Hitman game launcher is not clickable

### ISSUE THREAD

**Issue by jtheoof** [swaywm/sway#4600]
* Sway Version:

sway version 1.2 (Sep 26 2019, Arch Linux)

* Debug Log:

No useful log in `journalctl`:

```
Sep 30 22:35:40 ansan /usr/lib/gdm-wayland-session[1650]: >>> Adding process 31201 for game ID 863550
Sep 30 22:35:40 ansan /usr/lib/gdm-wayland-session[1650]: >>> Adding process 31213 for game ID 863550
Sep 30 22:35:40 ansan /usr/lib/gdm-wayland-session[1650]: Fontconfig warning: "/etc/fonts/conf.avail/05-reset-dirs-sample.conf", line 6: unknown element "reset-dirs"
Sep 30 22:35:41 ansan /usr/lib/gdm-wayland-session[1650]: ERROR: ld.so: object '/home/jattali/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
Sep 30 22:35:41 ansan /usr/lib/gdm-wayland-session[1650]: Game update: AppID 863550 "", ProcID 31184, IP 0.0.0.0:0
Sep 30 22:35:41 ansan /usr/lib/gdm-wayland-session[1650]: Setting breakpad minidump AppID = 863550
Sep 30 22:35:41 ansan /usr/lib/gdm-wayland-session[1650]: Steam_SetMinidumpSteamID:  Caching Steam ID:  76561198001543971 [API loaded no]
Sep 30 22:35:41 ansan /usr/lib/gdm-wayland-session[1650]: ERROR: ld.so: object '/home/jattali/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
Sep 30 22:35:41 ansan /usr/lib/gdm-wayland-session[1650]: ERROR: ld.so: object '/home/jattali/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
Sep 30 22:35:41 ansan /usr/lib/gdm-wayland-session[1650]: Game update: AppID 863550 "", ProcID 31229, IP 0.0.0.0:0
Sep 30 22:35:41 ansan /usr/lib/gdm-wayland-session[1650]: Setting breakpad minidump AppID = 863550
Sep 30 22:35:41 ansan /usr/lib/gdm-wayland-session[1650]: Steam_SetMinidumpSteamID:  Caching Steam ID:  76561198001543971 [API loaded no]
Sep 30 22:35:41 ansan /usr/lib/gdm-wayland-session[1650]: >>> Adding process 31220 for game ID 863550
Sep 30 22:35:41 ansan /usr/lib/gdm-wayland-session[1650]: >>> Adding process 31222 for game ID 863550
Sep 30 22:35:41 ansan /usr/lib/gdm-wayland-session[1650]: >>> Adding process 31229 for game ID 863550
```

* Configuration File: 

Nothing specific (IMO).

Finally, explain the steps you took in plain English to reproduce the problem below. Simply launch [Hitman](https://store.steampowered.com/app/236870/HITMAN/) through Steam. The game launcher appears but none of the 3 buttons are clickable. It's as if the window does not exist. I tried to look for it using `swaymsg -t get_tree` but I cannot find it. Any interaction with the buttons simply go through and are received by whatever window is underneath.

See screenshot.

Not sure if this is as sway bug or a steam bug. Sorry for the noise if I put this in the wrong place.

![Screenshot-2019-09-30-223210](https://user-images.githubusercontent.com/90020/65930421-b7a29180-e3d3-11e9-87a1-b7d731ebda21.png)
 

* Stack Trace, if sway crashes:

No crash

**Comment by jtheoof** [comment]
It would be related to #1865 though, even though it's not the same UI as Steam at this point.  `ps` points to this:

```
❯ ps -ef | rg HITMAN
jattali    32563   32562  0 22:48 ?        00:00:00 python3 /game/jattali/Steam/steamapps/common/Proton 4.11/proton waitforexitandrun /game/jattali/Steam/steamapps/common/HITMAN2/Launcher.exe
jattali    32568   32563  0 22:48 ?        00:00:00 steam /game/jattali/Steam/steamapps/common/HITMAN2/Launcher.exe
jattali    32614   32562  0 22:48 ?        00:00:00 Z:\game\jattali\Steam\steamapps\common\HITMAN2\Launcher.exe
```

**Comment by Wychmire** [comment]
This occurs with the LOTRO launcher and the EVE online launchers as well, when run from Steam or just using standard WINE. To play these games I run dwm since they don't work with Sway.

The LOTRO launcher I know uses Qt4, maybe it has something to do with that?

Unsure about Hitman, but the LOTRO and EVE launchers will follow you through workspaces in addition to being unclickable/unmovable

**Comment by Lucaber** [comment]
Same issue with Trackmania and other games, the mouse seams to be stuck on the right of the screen. I can move the courser up and down, but not to the left/right.

**Comment by aqxa1** [comment]
I have the same problem. As a workaround, I adapted [flibitijibibo's Shenmue 1 & 2 workaround](https://github.com/ValveSoftware/Proton/issues/1981#issuecomment-450759700) to work with Hitman 2:

```
#include <windows.h>

int main(int argc, char **argv)
{
        const char *name, *dir;
        STARTUPINFO startup;
        PROCESS_INFORMATION process;

        name = "Retail/HITMAN2.exe";
        dir = "Retail";

        ZeroMemory(&startup, sizeof(startup));
        ZeroMemory(&process, sizeof(process));
        startup.cb = sizeof(startup);
        if (CreateProcess(
                name,
                NULL,
                NULL,
                NULL,
                TRUE,
                0,
                NULL,
                dir,
                &startup,
                &process
        )) {
                WaitForSingleObject(process.hProcess, INFINITE);
                CloseHandle(process.hProcess);
                CloseHandle(process.hThread);
        }
        return 0;
}
```

Assuming you have Wine installed:

- Backup old Launcher.exe in Hitman 2 directory
- `winegcc launcher.c -o Launcher.exe`
- Copy Launcher.exe and Launcher.exe.so into Hitman 2 directory

You can also compile with mingw if you have that installed. The only difference is that there won't be a Launcher.exe.so to copy.

This will bypass the launcher entirely, allowing the game to work until the bug is fixed.

**Comment by billop** [comment]
Isn't this related to #4857? Tomb Raider is another game by Feral that has issues as well. The launcher is clickable, but the game cannot fullscreen properly.

**Comment by aqxa1** [comment]
Tomb Raider (2013)? Anyway in #4857 the issues on my system can be worked around by various methods that don't involve altering the game, but with Hitman 2, nothing but completely bypassing the launcher works. So it's possible that the issue here is different.

**Comment by rhysperry111** [comment]
@aqxa1 Do you have any advice for adapting the fix to work with FSX? I have attached a screenshot of the game's directory
![screenshot@2020 05 23-18:02:38](https://user-images.githubusercontent.com/28820694/82736209-c30a4a80-9d1f-11ea-9f68-bb7d6ad78ac1.png)

**Comment by aqxa1** [comment]
@rhysperry111 It's a bit hard to tell just from that screenshot. Do you know which executable is the launcher itself?

**Comment by rhysperry111** [comment]
@aqxa1 I believe it is fsx.exe

**Comment by aqxa1** [comment]
@rhysperry111 Okay, it then depends on which of those other executables are the actual game or games (the FS*.exe files, presumably). Perhaps something like:
```
name = "FS2002.exe"
dir = "."
```

and then compile launcher.c as `winegcc -c launcher.c -o fsx.exe`.  Maybe try setting dir to "NULL" if it doesn't work (which will also use the current directory).

**Comment by rhysperry111** [comment]
@aqxa1 It did not work with any of the .exe files in the game's directory. I'm assuming this means it would require a more complicated fix

**Comment by aqxa1** [comment]
@rhysperry111 Yeah, that's probably the case. I just tested XCom 2 and couldn't get that too work (it seems to have multiple launcher files located across several directories). The game itself might need additional parameters too, and can't just be launched as a bare exe file.

I'd look at that SteamEdit util as well.

**Comment by junglerobba** [comment]
So apparently the issue is views that start as override-redirect and then switch to normal views at runtime.
This exact problem was recently fixed in Wayfire with this: https://github.com/WayfireWM/wayfire/pull/706
How applicable is that fix to sway?

**Comment by emersion** [comment]
Ah, right, I don't think we support that. We'd need to destroy/create views after they're mapped.

**Comment by microo8** [comment]
I don't have the hitman game, but skyrim and fallout4 launchers are also not clickable.
Even though I have sway version 1.5-a7414885 which has the #5650 pr.

**Comment by sunng87** [comment]
I installed from sway-git in AUR and confirm the issue was fixed for Armored Brigade

**Comment by microo8** [comment]
It still doesn't work for me. `sway version 1.5-63420a2c (Feb  8 2021, branch 'master')`
